pyproject.toml

```diff
@@ -286,6 +286,51 @@ extend-exclude = [
 "great_expectations/optional_imports.py" = [
     "TID251",  # flake8-banned-api
 ]
+"great_expectations/dataset/*" = [ # dataset is deprecated
+    "TID251",  # flake8-banned-api
+]
+
+# #########################################
+# BEGIN temporary banned-api exclusion list
+# TODO: remove packages from this list as they are addressed
```

Comment:
We could also just make ruff add `noqa` comments for all these violations.
But then we would be touching a lot of files.